engine/Divert.js

```diff
@@ -27,24 +28,31 @@ export class Divert extends InkObject{
 	get targetPath(){
 		// Resolve any relative paths to global ones as we come across them
 		if (this._targetPath != null && this._targetPath.isRelative) {
-			var targetObj = this.targetContent;
+			var targetObj = this.targetPointer.Resolve();
 			if (targetObj) {
 				this._targetPath = targetObj.path;
 			}
 		}
-		
+
 		return this._targetPath;
 	}
 	set targetPath(value){
 		this._targetPath = value;
-		this._targetContent = null;
+		this._targetPointer = Pointer.Null;
 	}
-	get targetContent(){
-		if (this._targetContent == null) {
-			this._targetContent = this.ResolvePath(this._targetPath);
+	get targetPointer(){
+		if (this._targetPointer.isNull) {
+			var targetObj = this.ResolvePath(this._targetPath).obj;
+
+			if (this._targetPath.lastComponent.isIndex) {
+				_targetPointer.container = targetObj.parent;
```

Comment:
On this and the next couple of lines, I think we want `this._targetPointer` instead of just `_targetPointer`.